src/solver/hydro/management/HydroErrorsCollector.cpp

```diff
@@ -0,0 +1,36 @@
+#include "antares/solver/hydro/management/HydroErrorsCollector.h"
+
+#include <yuni/core/logs.h>
+
+#include "antares/antares/fatal-error.h"
+
+namespace Antares
+{
+void HydroErrorsCollector::Collect(const std::string& area_name, const std::string& message)
+{
+    logs.error() << "In Area " << area_name << " " << message;
+    area_errors_counter_[area_name]++;
+    errors_limit_reached_ = area_errors_counter_[area_name] > 10;
+    stop_ = true;
+
+    if (errors_limit_reached_)
+    {
+        throw FatalError("Hydro validation has failed !");
+    }
+}
+
+void HydroErrorsCollector::Collect(const std::string& message)
+{
+    logs.error() << message;
+    stop_ = true;
+}
+
+void HydroErrorsCollector::CheckForFatalErrors() const
+{
+    if (stop_ || errors_limit_reached_)
+    {
+        throw FatalError("Hydro validation has failed !");
+    }
```

Comment:
We should `throw` if any error have been encoutered, not depending on a threshold.

Reply:
this method is called after all hydro checks (case max {errors for all area} <10)